tests/Unit/Scraper/Application/ConfiguratorTest.php

```diff
@@ -215,10 +215,11 @@ public function whenTryToFindNewXpathButNotFoundItShouldLogItAndResetVariant():
             
         try {
             $this->configurator->configureFromDataset($posts);
-            Event::assertDispatched(ConfigurationScraped::class);
         } catch (ConfigurationException $e) {
             self::assertEquals('Field(s) ":field-1:" not found.', $e->getMessage());
         }
+
+        Event::assertDispatched(ConfigurationScraped::class);
```

Comment:
Almost! Please follow the example adding the assertions in the catch:
```
try {
   // Execution
   // Fail assertion
}
catch( <Exception> ) {
   // Asserts
}
```

And it can be even improved adding the fail assertion if it not launch the Exception.
```suggestion
            $this->fail('A ConfigurationException should be thrown')
        } catch (ConfigurationException $e) {
            Event::assertDispatched(ConfigurationScraped::class);
            self::assertEquals('Field(s) ":field-1:" not found.', $e->getMessage());
        }
```

The same should be applied for the other tests with the same schema.